spec/models/agents/weather_agent_spec.rb

```diff
@@ -29,10 +43,58 @@
     expect(agent).to be_valid
     expect(agent.working?).to be_falsey
   end
-  
+
+  context "wunderground" do
+    it "validates the location properly" do
+      expect(agent.options["location"]).to eq 94103
+      agent.options["location"] = "meh, really whatever, this is on the way to deprecation anyway"
+      expect(agent).to be_valid
+      agent.options["location"] = nil
+      expect(agent).to_not be_valid
+      agent.options["location"] = 94103
+    end
+  end
+
+  context "dark sky" do
+    it "validates the location properly" do
+      expect(dark_sky_agent.options["location"]).to eq "37.779329,-122.41915"
+      expect(dark_sky_agent).to be_valid
+      dark_sky_agent.options["location"] = "37.779329, -122.41915" # with a space
+      expect(dark_sky_agent).to be_valid
+      dark_sky_agent.options["location"] = "94103" # a zip code
+      expect(dark_sky_agent).to_not be_valid
+      dark_sky_agent.options["location"] = "37.779329,-122.41915"
+      expect(dark_sky_agent.options["location"]).to eq "37.779329,-122.41915"
+      expect(dark_sky_agent).to be_valid
+    end
+    it "fails cases that pass the first test but are invalid" do
+        dark_sky_agent.options["location"] = "137.779329, -122.41915" # too high latitude
```

Comment:
This indentation looks off

Reply:
ugh my text editor was being difficult, sorry I missed that.